lib/rules/selector-not-notation/index.js

```diff
@@ -110,10 +110,13 @@ const rule = (primary, _, context) => {
 						if (mustFix) return fixSimple(pseudo);
 					}
 
+					assert(pseudo.source && pseudo.source.end);
+
 					report({
 						message: messages.expected(primary),
 						node: ruleNode,
 						index: pseudo.sourceIndex,
+						endIndex: pseudo.source.end.column - 1,
```

Comment:
[ask] Should `endColumn` be exclusive like other rules?

```suggestion
						endIndex: pseudo.source.end.column,
```

Reply:
@ybiquitous AFAIK endIndex starts at 0 but column starts at 1 hence the -1.

e.g. for
`code: ':not(:not()) {}',`
I want to include `:not(:not())` and not `:not(:not()) `, correct?

Reply:
In the PostCSS API, `endColumn` means ["exclusive end position"](https://postcss.org/api/#warning-endcolumn). See below:

```css
:not(:not()) {}
/**         ↑    
↑-----------↑
1           13 (exclusive)
*/
```

In other rules, `endColumn` is exclusive, for example:

```css
a { opacity: 10% }
/**          ↑  ↑
             ↑--↑
             14 17
*/
```

https://github.com/stylelint/stylelint/blob/9802b9de2a40eb682df141ef2f121ade1631843c/lib/rules/alpha-value-notation/__tests__/index.js#L65-L71